File: Homework 3/Frontend/src/models/WineriesFilter.ts
export class WineriesFilter {
    public searchTerm: string | null = null;
    
    public ratings: number[] = [];

    public locations: string[] = [];

    public maxEntries: number | null = null;

    constructor(searchTerm: string | null, ratings: number[], locations: string[], maxEntries: number | null = null) {
        this.searchTerm = searchTerm;
        this.ratings = ratings;
        this.locations = locations;
        this.maxEntries = maxEntries;
    }
}

File: Homework 3/Backend/src/WineriesApp.Services/Models/Filters/WineriesFilter.cs
namespace WineriesApp.Services.Models.Filters
{
    public class WineriesFilter
    {
        public string? SearchTerm { get; set; }
    
        public double[] Ratings { get; set; } = Array.Empty<double>();

        public Guid[] Locations { get; set; } = Array.Empty<Guid>();
        
        public int? MaxEntries { get; set; }
    }
}